### PR TITLE
Add thread-factory for fork-join-pool threads

### PR DESCRIPTION
**Pull request by ggezer**
This PR introduces a thread-factory implementation for use by `ForkJoinPool` in order to provide a custom name for worker threads.

### CODE REVIEW

**Comment by gpang** [comment]
@ggezer Could you add a PR description for this?

**Comment by ggezer** [comment]
jenkins, test this please

**Comment by ggezer** [comment]
jenkins, test this please

File: core/common/src/main/java/alluxio/concurrent/jsr/ForkJoinPool.java [hunk 1]
```diff
@@ -2495,6 +2496,45 @@ public ForkJoinWorkerThread run() {
...
+   * Default ForkJoinWorkerThreadFactory implementation; creates a new ForkJoinWorkerThread using
+   * the system class loader as the thread context class loader.
+   */
+  public static final class AlluxioForkJoinWorkerThreadFactory
+          implements ForkJoinWorkerThreadFactory {
+    private static final AccessControlContext ACC = contextWithPermissions(
+            // new RuntimePermission("setContextClassLoader"), // java9-concurrent-backport changed
+            new RuntimePermission("getClassLoader"));
+    // Thread id counter.
+    private static final AtomicLong sThreadId = new AtomicLong(0);
```
Comment (gpang) [inline core/common/src/main/java/alluxio/concurrent/jsr/ForkJoinPool.java:2509]: minor question, do thread ids typically start from 0 or 1?

Comment (ggezer) [reply core/common/src/main/java/alluxio/concurrent/jsr/ForkJoinPool.java:2509]: That's how guava thread-factory that we use assign indices to threads.

Comment (ggezer) [reply core/common/src/main/java/alluxio/concurrent/jsr/ForkJoinPool.java]: changed the name to better reflect what it's an index.

Comment (gpang) [reply core/common/src/main/java/alluxio/concurrent/jsr/ForkJoinPool.java]: ok, sounds good

**Comment by ggezer** [comment]
alluxio-bot, merge this please

**Comment by ZacBlanco** [comment]
alluxio-bot, merge this please

**Comment by ggezer** [comment]
alluxio-bot, cherry-pick this to branch-2.3 please

**Pull request merged**